cpp/include/kvikio/posix_io.hpp

```diff
@@ -183,18 +206,20 @@ std::size_t posix_device_io(int fd,
  * If `size` or `file_offset` isn't aligned with `page_size` then
  * `fd` cannot have been opened with the `O_DIRECT` flag.
  *
+ * @tparam ioDataCompletionLevel Whether all requested data are processed or not. If `FULL`, all of
+ * `count` bytes are read.
  * @param fd File descriptor
  * @param buf Base address of buffer in host memory.
  * @param size Size in bytes to read.
  * @param file_offset Offset in the file to read from.
- * @param partial If false, all of `size` bytes are read.
  * @return Size of bytes that were successfully read.
  */
-inline std::size_t posix_host_read(
```

Comment:
This change and the one to write are breaking. Is this acceptable? @madsbk

Reply:
Watching this, to see how breaking changes are best handled.  (One option would have been to deprecate in one release, and then remove in the other.)

Reply:
I agree, a deprecation cycle is a good thing however, it is unlikely that anyone is calling this function directly. At least, a search on github [doesn't show anything](https://github.com/search?q=posix_host_read&type=code).

In this case, I think we should allow a breaking chance without deprecation. In fact, I think we should put all of the functions in `posix_io.hpp` under `kvikio::detail`, they are not meant to be used directly.